src/appleseed.studio/mainwindow/renderingsettingswindow.cpp

```diff
@@ -652,6 +662,59 @@ namespace
         }
     };
 
+    class InteractiveImagePlaneSamplingPanel
+      : public ImagePlaneSamplingPanel
+    {
+        Q_OBJECT
+
+      public:
+        InteractiveImagePlaneSamplingPanel(const Configuration& config, QWidget* parent = nullptr)
+          : ImagePlaneSamplingPanel(parent)
+        {
+            QVBoxLayout* layout = new QVBoxLayout();
+            container()->setLayout(layout);
+
+            create_image_plane_sampling_general_settings(layout);
+
+            load_general_sampler(config);
+        }
+
+        void save_config(Configuration& config) const override
+        {            
+            if (get_widget<bool>("general.unlimited_samples"))
+                config.get_parameters().remove_path("progressive_frame_renderer.max_samples");
+            else set_config(config, "progressive_frame_renderer.max_samples", get_widget<int>("general.max_samples"));
+        }
+
+      private:
+        void create_image_plane_sampling_general_settings(QVBoxLayout* parent)
+        {
+            QGroupBox* groupbox = new QGroupBox("General");
```

Comment:
`Progressive` might be a more precise name for this group, however this seems implied by the configuration title, `interactive`. Maybe a group here isn't even worthwhile.

Reply:
Indeed maybe we can get rid of the group.

Are these settings only exposed in interactive configurations?

Reply:
This change causes this specific setting to be exposed in interactive mode only.

Looking at the ui, there are other settings related to max samples exposed per the different sampler types in final mode, but none currently for max total number of samples.

Reply:
> This change causes this specific setting to be exposed in interactive mode only.

Ok, that's fine then.

> Looking at the ui, there are other settings related to max samples exposed per the different sampler types in final mode, but none currently for max total number of samples.

I'm not sure to understand what you mean here?

Reply:
I think I'm just repeating in saying that max samples is only exposed in interactive mode.

Reply:
(sorry for the confusion, my understanding of the project is still growing, thus the comment in question seems naïve)

Reply:
No problem!